src/app/ReviewService/ReviewService.cs

```diff
@@ -15,33 +17,75 @@ public sealed class ReviewService<TReviewSettings> : IReviewService<TReviewSetti
 	private readonly IReviewPrompter _reviewPrompter;
 	private readonly IReviewSettingsSource<TReviewSettings> _reviewSettingsSource;
 	private readonly ReviewConditionCallback<TReviewSettings>[] _reviewConditions;
+	private readonly ILogger _logger;
 
 	public ReviewService(
+		ILogger<ReviewService<TReviewSettings>> logger,
 		IReviewPrompter reviewPrompter,
 		IReviewSettingsSource<TReviewSettings> reviewSettingsSource,
 		ReviewConditionCallback<TReviewSettings>[] reviewConditions
 	)
 	{
+		_logger = logger;
 		_reviewPrompter = reviewPrompter;
 		_reviewSettingsSource = reviewSettingsSource;
 		_reviewConditions = reviewConditions;
 	}
 
 	/// <inheritdoc/>
-	public Task TryRequestReview(CancellationToken ct)
+	public async Task TryRequestReview(CancellationToken ct)
 	{
-		throw new NotImplementedException();
+		_logger.LogDebug("Trying to request a review.");
+
+		if (await GetAreConditionsSatisfied(ct))
+		{
+			_logger.LogInformation("Conditions were satisfied. Requesting a review.");
+			await _reviewPrompter.TryPrompt();
+		}
+		else
+		{
+			_logger.LogWarning("Failed to request a review because one or more conditions were not satisfied.");
+		}
 	}
 
 	/// <inheritdoc/>
-	public Task<bool> GetAreConditionsSatisfied(CancellationToken ct)
+	public async Task<bool> GetAreConditionsSatisfied(CancellationToken ct)
 	{
-		throw new NotImplementedException();
+		_logger.LogDebug("Evaluation conditions");
+
+		var currentSettings = await _reviewSettingsSource.Read(ct);
+		var reviewConditionTasks = _reviewConditions.Select(async condition => await condition(ct, currentSettings, DateTimeOffset.Now));
+
+		var result = (await Task.WhenAll(reviewConditionTasks)).All(x => x is true);
+
+		if (result)
+		{
+			_logger.LogInformation("Evaluated conditions and all conditions are satisfied.");
+		}
+		else
+		{
+			_logger.LogWarning("Evaluted conditions and one or more conditions were not satisfied.");
```

Comment:
Same here, I wouldn't use warning because it's normal that conditions are not always met.